cmd/determinize-ci-operator/main.go

```diff
@@ -103,6 +110,56 @@ func main() {
 	}
 }
 
+func upgradeWorkflowForClusterProfile(clusterProfile api.ClusterProfile) string {
+	return fmt.Sprintf("openshift-upgrade-%s-loki", clusterProfile)
+}
+
+func e2eWorkflowForClusterProfile(clusterProfile api.ClusterProfile) string {
+	return fmt.Sprintf("openshift-e2e-%s", clusterProfile)
+}
+
+func migrateOpenShiftInstallerTemplates(
+	configuration *config.DataWithInfo,
+	allowedBranches sets.String,
+	allowedOrgs sets.String,
+	allowedCloudproviders sets.String,
+) (migratedCount int) {
+	if (len(allowedBranches) != 0 && !allowedBranches.Has(configuration.Info.Branch)) || (len(allowedOrgs) != 0 && !allowedOrgs.Has(configuration.Info.Org)) {
+		return 0
+	}
+
+	for idx, test := range configuration.Configuration.Tests {
+		if test.OpenshiftInstallerClusterTestConfiguration == nil ||
+			(len(allowedCloudproviders) != 0 && !allowedCloudproviders.Has(string(test.OpenshiftInstallerClusterTestConfiguration.ClusterProfile))) {
+			continue
+		}
+
+		clusterProfile := test.OpenshiftInstallerClusterTestConfiguration.ClusterProfile
+		switch {
+		case test.OpenshiftInstallerClusterTestConfiguration.Upgrade:
+			test.OpenshiftInstallerClusterTestConfiguration = nil
+			test.MultiStageTestConfiguration = &api.MultiStageTestConfiguration{
+				ClusterProfile: clusterProfile,
+				Workflow:       utilpointer.StringPtr(upgradeWorkflowForClusterProfile(clusterProfile)),
+			}
+		case test.Commands == "setup_ssh_bastion; TEST_SUITE=openshift/disruptive run-tests; TEST_SUITE=openshift/conformance/parallel run-tests":
+			// TODO(muller): Unfortunately there is no easy way to express this ("run same step twice")
```

Comment:
Technically what's missing is passing separate parameters to different invocations of a single ref, right? You can run the same step as many times as you need. Any way to just get `openshift-tests` to understand two test suites via one invocation?

Reply:
Yes, passing separate parameters to multiple invocations of a single ref is exactly what we'd need. I'm not sure if same step can be ran multiple times (wouldn't the resulting `Pods` have same names, leading to some kind of false cache ("this step was already run == this pod already exists") hit?

I considered adding a parameter to the step that would be a list of suites, and the step would iterate over them. I'm worried about the step timeout that we have on the step though.